File: hibernate-scala-examples/src/main/scala/org/hibernate/examples/hibernate/config/AbstractH2HibernateConfiguration.scala
package org.hibernate.examples.hibernate.config

import java.util.Properties
import javax.sql.DataSource
import org.hibernate.cfg.AvailableSettings
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.transaction.annotation.EnableTransactionManagement
import org.hibernate.examples._

/**
 * HSQL DB를 사용하는 Spring 용 Hibernate 환경설정 정보
 *
 * @author 배성혁 sunghyouk.bae@gmail.com
 * @since 2013. 12. 24. 오후 4:56
 */
@Configuration
@EnableTransactionManagement
abstract class AbstractH2HibernateConfiguration extends AbstractHibernateConfiguration {

    @Bean
    override def dataSource(): DataSource =
        buildDataSource(DRIVER_CLASS_H2,
                           "jdbc:h2:mem:" + getDatabaseName + ";MVCC=TRUE;",
                           "sa",
                           "")

    override def hibernateProperties(): Properties = {
        val props = super.hibernateProperties()
        props.put(AvailableSettings.DIALECT, DIALECT_H2)
        props
    }
}

File: hibernate-examples/src/main/java/org/hibernate/examples/hibernate/config/AbstractH2HibernateConfiguration.java
package org.hibernate.examples.hibernate.config;

import org.hibernate.cfg.Environment;
import org.springframework.context.annotation.Bean;

import javax.sql.DataSource;
import java.util.Properties;

/**
 * AbstractH2HibernateConfiguration
 *
 * @author sunghyouk.bae@gmail.com
 * @since 2014. 2. 28.
 */
public abstract class AbstractH2HibernateConfiguration extends AbstractHibernateConfiguration {

  public static final String DRIVER_CLASS_H2 = "org.h2.Driver";
  public static final String DIALECT_H2 = "org.hibernate.dialect.H2Dialect";

  @Bean
  @Override
  public DataSource dataSource() {
    return buildDataSource(DRIVER_CLASS_H2,
        "jdbc:h2:mem:" + getDatabaseName() + ";DB_CLOSE_DELAY=-1;MVCC=TRUE;",
        "sa",
        "");
  }

  @Override
  public Properties hibernateProperties() {
    Properties props = super.hibernateProperties();
    props.put(Environment.DIALECT, DIALECT_H2);
    return props;
  }
}
